ramips: update board support for WIZnet WizFi630S

7tobias opened the pull request:
This PR is needed as in the release version of WizFi630S some pins had changed that where not yet updated in master. Also in the factory image a different board name is used, which caused a incompatibility warning when using sysupgrade. This PR is correcting the pin map and is updating the supported device list to avoid the error. There's also some cosmetics (removing a doublet). WizFi630S also has three mac adresses preprogrammed in the factory partion, but did not use it by now.

adrianschmutzler commented:
"factory image" typically refers to an OpenWrt image built for flashing over vendor firmware. I suspect you mean the OEM/vendor firmware when you refer to "factory image"?

adrianschmutzler commented:
Hmm, your initial support commit only mentions tftp flashing:
https://github.com/openwrt/openwrt/commit/7c4b85ee278576816b0f9d20e48fa696f8793fdf

So, does your first patch actually enable flashing directly from vendor firmware? If yes, please make that more obvious in the commit message.

Apart from that, note that your commit title/message have formal issues. Please check out https://openwrt.org/submitting-patches again and check for the maximum line length of title and message.

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -23,12 +23,13 @@
 		serial1 = &uartlite;
 	};
 
+
```
Please don't add all these empty lines.

7tobias replied:
ups - sorry.

> > WizFi630S had some pins changed in the release version of the board. The run led, wps led and wps button where affected. This patch is correcting this.
> 
> Sure? Why has nobody complained so far? Maybe this is just another revision?

Sure - this pinout is the only one used in mass production.

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -172,13 +154,10 @@
 
 &esw {
 	mediatek,portmap = <0x3e>;
+	mediatek,portdisable = <0x26>;
```
That's not a pin change.

7tobias replied:
WizFi630S is using only 3 phy ports - so the unneded 2 are disabled to save power. I'll put it to a separate commit.

adrianschmutzler replied:
Yes, but IIRC the driver only takes three hard-coded values, 0x3e, 0x2f and a third one I don't remember.

adrianschmutzler replied:
https://github.com/openwrt/openwrt/pull/2667

7tobias replied:
> Yes, but IIRC the driver only takes three hard-coded values, 0x3e, 0x2f and a third one I don't remember.

https://github.com/openwrt/openwrt/pull/2667 is basically telling that mt7530 is accepting only three values. 

From what i know it's a binary map. `root@OpenWrt:/# swconfig dev switch0  show | grep Port -A 1
Port 0:
        disable: 0
--
Port 1:
        disable: 1
--
Port 2:
        disable: 1
--
Port 3:
        disable: 0
--
Port 4:
        disable: 0
--
Port 5:
        disable: 1
--
Port 6:
        disable: 0
root@OpenWrt:/#
` At runtime the portmap resolves to the expected values in the switchconfig. The module is using Port 0, 3, 4 and 6 of course.

mans0n replied:
The hard-coded one is the portmap for mt7530. The portdisable property is a binary map, right.

adrianschmutzler replied:
Sorry, I mixed up portmap and portdisable here, didn't look properly. Thanks for verifying on device nevertheless!

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -172,13 +154,10 @@
-
-&wmac {
-	status = "okay";
-};
```
That's not a pin change.

7tobias replied:
This was just an obvious doublet in the dts file - so it's just a cosmetic change - i can put it to a separate commit.

> "factory image" typically refers to an OpenWrt image built for flashing over vendor firmware. I suspect you mean the OEM/vendor firmware when you refer to "factory image"?

yes you are right - i'll change this

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -81,27 +83,7 @@
-
-	refclk {
-		groups = "refclk";
+		groups = "gpio", "wdt", "p1led_an", "p2led_an", "i2s", "i2c", "spis", "pwm0", "pwm1", "refclk", "perst";
```
I don't think you will need all of those ...

mans0n replied:
You enabled i2c but disabled i2c pins. Does the i2c work with this setting?

7tobias replied:
> You enabled i2c but disabled i2c pins. Does the i2c work with this setting?
Most likely it will not - i wanted to keep  compatibility with the buttons and leds from the old evaluation board (the module works also fine in it) which has shared pins with i2c and i2s - i think i'll have to move the data to a dtsi file and make separate dts files for both eval boards.

7tobias replied:
i2c_sd is a shared pin with the run (system led) on the eval boards. So it's a either or. Should i2c then be enabled or not?

> You enabled i2c but disabled i2c pins. Does the i2c work with this setting?

adrianschmutzler replied:
Based on leds and keys I only get the following here:
i2c p1led_an i2s gpio p2led_an

I don't think we should anything else without a specific reason.

7tobias replied:
pwm0, pwm1 and uart2 pins are also available. These are not labelled explicitly, but available on pin headers on the evalboard. Should these then be enabled?

mans0n replied:
If the feature can be used (Have you tested pwm and uart BTW?), enable the feature node. If the pin is used for other purpose (like led), then disable the feature node and set the pin as gpio.

7tobias replied:
In one point I was wrong, the pins are just on the module, but not on the evalation board. So I cannot test without a customized mother board.

adrianschmutzler commented:
> WizFi630S had some pins changed in the release version of the board. The run led, wps led and wps button where affected. This patch is correcting this.

Sure? Why has nobody complained so far? Maybe this is just another revision?

mans0n commented:
1. You can remove `#address-cells` and `#size-cells` properties in the `keys` node as well.
2. `m25p,chunked-io = <31>;` line can be removed according to the commit be2b61e4f1ec.
    It seems all current usages are added after the said commit (probably new device patches had not been rebased before merged)

adrianschmutzler commented:
> 2\. `m25p,chunked-io = <31>;` line can be removed according to the commit [be2b61e](https://github.com/openwrt/openwrt/commit/be2b61e4f1ec1e351733bebb7105629599034639).
>     It seems all current usages are added after the said commit (probably new device patches had not been rebased before merged)

I've sent a patch for that.

7tobias committed:
ramips: correct WizFi630S pin mappings

WizFi630S had some pins changed in the release version of the board.
The run led, wps button and a slide switch where affected.
This patch is correcting this.
i2c is removed as it is sharing a pin with the run (system) led.
uart2 is enabled as it is also enabled in the OEM firmware.

Signed-off-by: Tobias Welz <tw@wiznet.eu>

7tobias committed:
ramips: WizFi630S use macs from factory partion

WIZnet WizFi630s has three mac addresses in the factory partition:
0x04 (also on the label), 0x28 for wan mac and 0x2e as lan mac.
All three macadresses are sequential series of addresses.
This is making use of them.

MAC addresses as verified by OEM firmware:

vendor   OpenWrt   source
WLAN     ra0       factory 0x04 (label)
WAN      eth0.2    factory 0x28 (label + 1)
LAN      eth0.1    eth0 + 1     (label + 2)

Signed-off-by: Tobias Welz <tw@wiznet.eu>

7tobias commented:
I made the changes, tested it and pushed the PR.

adrianschmutzler commented:
> ramips: add sysupgrade support to WizFi630S board.

One should add a comment that config should not be kept.

I'd personally use a different, more specific title, something like "ramips: enable flashing WizFi630S board via OEM firmware", if that's too long remove the "board" ...

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -61,13 +61,13 @@
 		scm1 {
 			label = "SCM1";
-			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
```
If that's a switch, we should add
`linux,input-type = <EV_SW>;`
?

7tobias replied:
Yes, two static sliders. I'll add it.

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -104,6 +104,16 @@
+	p2led_an {
+		groups = "p2led_an";
+		function = "gpio";
+	};
```
If we are touching this anyway, I'd prefer the syntax with one block as in your first attempt, just with the reduced number of entries. That one is used for the majority of devices. (Unless there is a specific reason for choosing this syntax here)

adrianschmutzler commented on target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:
```diff
@@ -200,7 +200,8 @@ ramips_setup_macs()
 	wavlink,wl-wn575a3|\
 	wiznet,wizfi630s)
-		wan_mac=$(macaddr_add "$(mtd_get_mac_binary factory 0x28)" 1)
+		wan_mac=$(mtd_get_mac_binary factory 0x4)
```
As stated in the other comment, please report what's the assignment on OEM firmware; maybe compile a table like here:
https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses
This can be added to the commit message then.

Apart from that, lan_mac is redundant here, as this is already done by the ethernet node in DTS.

BTW, apart from those points, note that you change the adresses for four independent devices here!

adrianschmutzler commented:
I've merged patch 2 and 3 for now.

adrianschmutzler commented on target/linux/ramips/mt76x8/base-files/etc/board.d/02_network:
```diff
@@ -201,7 +201,8 @@ ramips_setup_macs()
 	wavlink,wl-wn575a3|\
 	wiznet,wizfi630s)
-		wan_mac=$(macaddr_add "$(mtd_get_mac_binary factory 0x28)" 1)
+		wan_mac=$(mtd_get_mac_binary factory 0x28)
```
So, 0x28 is *:f1? Or is it *:f2?

7tobias replied:
The OEM firmware is not consistant during initalisation (lan / wan mac change). Maybe it's related to the fact that in the oem firmware the wan adress is derived as eth0 mac + 1 which is a "don't" according to the link you sent.
Here you can see three screenshots.
There are three columns: This Patch / OEM fully booted / OEM during init
![image](https://user-images.githubusercontent.com/2125300/89473369-f4bd7a00-d782-11ea-84a4-61b573dcbf41.png)
Anyway with this LAN / WAN have the same mac adresses.

adrianschmutzler commented on target/linux/ramips/dts/mt7628an_wiznet_wizfi630s.dts:
```diff
@@ -150,18 +132,10 @@
-&pwm {
-	status = "okay";
-};
-
```
Why is this one removed?

7tobias replied:
it was mentioned somewhere before, that if i cannot test the functionality it should not be in the feature node. As the pin is only available with someones own mother board, i currently have no possibility to test it, as it is not available on the evalboard. Normally i would leave it in, because i'd assume it should work.

adrianschmutzler replied:
Since this is about changing an existing state, I'd only _change_ it when verified.

7tobias committed:
ramips: enable flashing WizFi630S via OEM firmware

WIZnet WizFi630s board name is written slightly different it its OEM
OpenWrt firmware. This causes an incompatibility warning during flashing
with sysupgrade. This patch is adding the vendor board name to the
supported devices list to avoid this warning. For initial flashing you
can use sysupgrade via command line or luci beside of TFTP.
Do not keep the OEM configuration during sysupgrade.

Signed-off-by: Tobias Welz <tw@wiznet.eu>

adrianschmutzler commented:
Thanks for cleaning this up, I've now merged the rest, so we should be done here.

adrianschmutzler commented:
I also plan to backport the changes (except the MAC address adjustments) to 19.07 in a few days:
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/staging-19.07

7tobias commented:
> I also plan to backport the changes (except the MAC address adjustments) to 19.07 in a few days:
> https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/staging-19.07

I did a quick testing of your staging tree on the module - seems to be ok.
